devtools/maven/src/main/java/io/quarkus/maven/TrackConfigChangesMojo.java

```diff
@@ -102,34 +112,64 @@ protected void doExecute() throws MojoExecutionException, MojoFailureException {
             return;
         }
 
         CuratedApplication curatedApplication = null;
         QuarkusClassLoader deploymentClassLoader = null;
         final ClassLoader originalCl = Thread.currentThread().getContextClassLoader();
-        Properties actualProps;
         final boolean clearPackageTypeSystemProperty = setPackageTypeSystemPropertyIfNativeProfileEnabled();
         try {
             curatedApplication = bootstrapApplication(launchMode);
-            deploymentClassLoader = curatedApplication.createDeploymentClassLoader();
-            Thread.currentThread().setContextClassLoader(deploymentClassLoader);
-
-            final Class<?> codeGenerator = deploymentClassLoader.loadClass("io.quarkus.deployment.CodeGenerator");
-            final Method dumpConfig = codeGenerator.getMethod("dumpCurrentConfigValues", ApplicationModel.class, String.class,
-                    Properties.class, QuarkusClassLoader.class, Properties.class, Path.class);
-            dumpConfig.invoke(null, curatedApplication.getApplicationModel(),
-                    launchMode.name(), getBuildSystemProperties(true),
-                    deploymentClassLoader, compareProps, targetFile);
+            if (prevConfigExists || dumpCurrentWhenRecordedUnavailable) {
+                final Path targetFile = getOutputFile(outputFile, launchMode.getDefaultProfile(), "-config-check");
+                Properties compareProps = new Properties();
+                if (prevConfigExists) {
+                    try (BufferedReader reader = Files.newBufferedReader(compareFile)) {
+                        compareProps.load(reader);
+                    } catch (IOException e) {
+                        throw new RuntimeException("Failed to read " + compareFile, e);
+                    }
+                }
+
+                deploymentClassLoader = curatedApplication.createDeploymentClassLoader();
+                Thread.currentThread().setContextClassLoader(deploymentClassLoader);
+
+                final Class<?> codeGenerator = deploymentClassLoader.loadClass("io.quarkus.deployment.CodeGenerator");
+                final Method dumpConfig = codeGenerator.getMethod("dumpCurrentConfigValues", ApplicationModel.class,
+                        String.class,
+                        Properties.class, QuarkusClassLoader.class, Properties.class, Path.class);
+                dumpConfig.invoke(null, curatedApplication.getApplicationModel(),
+                        launchMode.name(), getBuildSystemProperties(true),
+                        deploymentClassLoader, compareProps, targetFile);
+            }
+
+            if (dumpDependencies) {
+                final List<String> deps = new ArrayList<>();
+                for (var d : curatedApplication.getApplicationModel().getDependencies(DependencyFlags.DEPLOYMENT_CP)) {
+                    StringBuilder entry = new StringBuilder(d.toGACTVString());
+                    if (d.isSnapshot()) {
```

Comment:
I decided to avoid computing the checksum if the dependency is not a snapshot one. They should be immutable in this case.

Reply:
Final versions could still be built locally. For example before releasing but not necessarily then.